### ISSUE TITLE
MQTT

### ISSUE THREAD

**Issue by evilpin** [guardianproject/haven#264]
MQTT support would be a very nice feature. It would allow for easy integration with most home automation platforms.

**Comment by lukeswitz** [comment]
This makes an even dozen requests for remote service interaction and for good reason: it's something we all want/need in the app. Any remote execution operations (talking back and forth) between devices is open to security issues.

That said, MQTT is super old and easy to hack. A simple search [provides recipes](https://morphuslabs.com/hacking-the-iot-with-mqtt-8edaf0d07b9b) (plus it's just too easy to MITM attack anything on the IoT). From my perspective, this project takes a zoomed out approach to security with an emphasis on human intent as well as simple interaction. I do agree though, remote controls are nice and we could start small using additional cameras.

**Comment by fat-tire** [comment]
That article is pretty much pointing out that people are running open MQTT servers (brokers) w/o encryption or really any authentication at all, but MQTT can be [used](https://www.hivemq.com/blog/mqtt-security-fundamentals-tls-ssl) w TLS. 

I do see how this could be kind of useful for someone wanting to check on many aspects of their device's physical status in real time.  The [paho](https://www.eclipse.org/paho/clients/android/) library for Android might be a good way to implement it.

**Comment by lukeswitz** [comment]
Right, if built correctly MQTT can be pretty secure. Does adding SSL on our end mean total privacy for the messages lifespan? 

Either way, we should consider adding a Snyk badge to the project. It takes minutes and points out things such as: https://snyk.io/vuln/npm:mqtt
